liota/transports/mqtt.py

```diff
@@ -48,25 +49,78 @@ def on_message(self, client, userdata, msg):
         self.client.on_publish = self.on_publish
         self.client.on_subscribe = self.on_subscribe
+        self.tls_support = tls_support
         self.connect_soc()
         TransportLayer.__init__(self)
 
     def connect_soc(self):
-        self.client.connect_soc(host=self.url, port=self.port, keepalive=60)
-        log.info("Connection Successful")
+        # Set up TLS support
+        if self.tls_support:
+
+            # Validate CA certificate path
+            if self.gw_identity.cacert:
+                if not(os.path.exists(self.gw_identity.cacert)):
+                    raise ValueError("Error : Wrong CA certificate path.")
+            else:
+                raise ValueError("Error : CA certificate path is missing")
+
+            # Validate client certificate path
+            if self.gw_identity.certfile:
+                if os.path.exists(self.gw_identity.certfile):
+                    client_cert_available = True
+                else:
+                    raise ValueError("Error : Wrong client certificate path.")
+            else:
+                client_cert_available = False
+
+            # Validate client key file path
+            if self.gw_identity.keyfile:
+                if os.path.exists(self.gw_identity.keyfile):
+                    client_key_available = True
+                else:
+                    raise ValueError("Error : Wrong client key path.")
+            else:
+                client_key_available = False
+
+            # Multiple conditions for certificate validations
+                # 1. Both Client certificate and key file should be present
+                # 2. If both are not there proceed without client certificate and key
+                # 3. If one of them is not there throw an error to notify one of them is missing
+            if client_cert_available != client_key_available:
+                raise ValueError("Error : Client cerificate or key is missing")
+            elif client_cert_available and client_key_available:
+                log.debug("Certificates : ", self.gw_identity.cacert, self.gw_identity.certfile, self.gw_identity.keyfile)
+
+                self.client.tls_set(self.gw_identity.cacert, self.gw_identity.certfile, self.gw_identity.keyfile,
+                                    cert_reqs=getattr(ssl, self.TLSConf.cert_required),
+                                    tls_version=getattr(ssl, self.TLSConf.tls_version), ciphers=self.TLSConf.cipher)
+            else:
+                self.client.tls_set(self.gw_identity.cacert,
+                                    cert_reqs=getattr(ssl, self.TLSConf.cert_required),
+                                    tls_version=getattr(ssl, self.TLSConf.tls_version), ciphers=self.TLSConf.cipher)
+
+            log.info("TLS support is set up.")
 
+        # Connect with MQTT Broker
+        self.client.on_connect = self.on_connect
+        self.client.connect(host=self.url, port=self.port, keepalive=60)
```

Comment:
Is keepalive not a param to be collected at method level? If yes, lets do that and have a default value.